logic/src/commonMain/kotlin/com/wire/kalium/logic/feature/message/RetryFailedMessageUseCase.kt

```diff
@@ -178,7 +178,10 @@ class RetryFailedMessageUseCase internal constructor(
 
             else -> handleError("Asset message with transfer status $assetTransferStatus cannot be retried")
         }
-            .onSuccess { retrySendingMessage(it) }
+            .onSuccess {
+                updateAssetMessageTransferStatus(AssetTransferStatus.UPLOADED, message.conversationId, message.id)
```

Comment:
Question: should the status be changed to `UPLOADED` before the message is actually uploaded? Maybe `UPLOAD_IN_PROGRESS` would be more appropriate and change to `UPLOADED` only after the successful resend 🤔

Reply:
I thought about that - but isnt that status only regarding the asset? We have the additional status for message `PENDING` etc. Also if we set up this status like that, we can see:
- Error sending
- Uploading
- Uploaded
- Message pending (until its sent)
- If there will be any connection problem the message will get back to failed state

Reply:
IMO, few line up is better place for this updating. 

```
return when (assetTransferStatus) {
    AssetTransferStatus.FAILED_UPLOAD, AssetTransferStatus.NOT_DOWNLOADED -> {
        updateAssetMessageTransferStatus(AssetTransferStatus.UPLOAD_IN_PROGRESS, message.conversationId, message.id)
        retryUploadingAsset(content)
            ...
            .onSuccess { updateAssetMessageTransferStatus(AssetTransferStatus.UPLOADED, message.conversationId, message.id) }
```

in that case we won't update status if it's already `UPLOADED` (less DB-queries - better :) )

Reply:
You are right! thanks, changed :)